integration-tests/test_satellite.py

```diff
@@ -0,0 +1,117 @@
+        cmd = ["hostnamectl", "status"]
+        output = subprocess.check_output(cmd, universal_newlines=True)
+        assert hostname_to_test in output
+
+        # redaction on hostname:
+        data_redaction(hostname_to_test)
+
+        # check the collection, if redaction pattern is filtered out
+        # collect data
+        archive_result = insights_client.run("--no-upload")
+
+        # extract the collected tar file
+        m = re.search("^Archive saved at (.+)$", archive_result.stdout, re.MULTILINE)
+        assert m
+        archive_location = m.group(1)
+        cmd = ["tar", "-vxzf", archive_location, "-C", tmp_path]
+        subprocess.check_call(cmd)
+
+        # the redaction data should not locate in the collected files
+        archive_extract = os.path.basename(archive_location).split(".")[0]
+        path_to_check = tmp_path / archive_extract
+        cmd = ["grep", "-rnw", ".", "-e", hostname_to_test]
+        # should not find the obfuscate pattern, the return code should not zero.
+        ret = subprocess.run(cmd, cwd=path_to_check)
+        assert ret.returncode == 1
+    finally:
+        # Recover the previous hostname for the system
+        set_hostname(previous_hostname)
+        # Remove all data redaction
+        os.remove(CONTENT_REDACTION_FILE)
+
+
+def test_set_ansible_host_info(insights_client):
+    """
+    Test if the ansible-host can be set with satellite
+    Related to https://issues.redhat.com/browse/RHELPLAN-160348,
+    and https://bugzilla.redhat.com/show_bug.cgi?id=222729
+    """
+    # Register system against Satellite, and register insights through satellite
+    insights_client.register()
+    sleep(10)
+    assert insights_client.is_registered
+
+    # Update ansible-host
+    ret = insights_client.run("--ansible-host=foo.example.com", check=False)
+    assert "Could not update Ansible hostname" not in ret.stdout
+    assert ret.returncode == 0
+
+
+@pytest.mark.parametrize("legacy_upload_value", [True, False])
+def test_register_group_option(insights_client, legacy_upload_value):
+    """
+    Bug https://issues.redhat.com/browse/RHINENG-7567 exists on both
+    production env and satellite env with legacy_upload=True.
+
+    With egacy_upload=False, "insights-client --register --group=tag"
+    works well on both envs.
+
+    We plan to compose test cases with legacy_upload=False,
+    but we also need to cover the existing bug,
+    so I check  legacy_upload=True to cover bug.
+    """
+    # make sure the system is not registered to insights
+    with contextlib.suppress(Exception):
+        insights_client.unregister()
+    assert not insights_client.is_registered
+    if legacy_upload_value is True:
+        cmd = ["insights-client", "--register", "--group=tag"]
+        subprocess.check_call(cmd, shell=True)
+        insights_client.unregister()
+    else:
+        insights_client.run("--register", "--group=tag")
+        insights_client.unregister()
```

Comment:
the parametrized value of `legacy_upload` is not actually configured for `insights-client`, which means it will run using its default (False); you need to manually set it:
```python
    insights_client.config.legacy_upload = legacy_upload_value
```

also, since the failing combination is `legacy_upload` + Satellite, then I'd do the following:
- unconditionally use the `run()` way of registering, also adding `check=False` and storing the command return
- in case the current environment is a Satellite one + the `legacy_upload` value is `False` + the test fails, use `pytest.xfail()` to properly mark the known failure situation as such

Reply:
actually, if it fails in any case when using `legacy_upload`, then I'd mark the `False` value of the `legacy_upload_value` parameter as xfail

Reply:
> the parametrized value of `legacy_upload` is not actually configured for `insights-client`, which means it will run using its default (False); you need to manually set it:

Yes, it is. At the same time, bug RHINENG-7567 happens when legacy_upload=True. 
So we can use pytest.xfail when legacy_upload=True. And run this test case with legacy_upload=False(default set in .secrets.toml) without specifying its value manually.